libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationViewListener.kt

```diff
@@ -0,0 +1,81 @@
+package com.mapbox.navigation.dropin
+
+import com.mapbox.geojson.Point
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.Style
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+
+/**
+ * Interface definition for the NavigationView listener.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+abstract class NavigationViewListener {
+    /**
+     * Called when destination has changed.
+     *
+     * @param destination destination Point or `null` when destination is cleared.
+     */
+    open fun onDestinationChanged(destination: Point?) = Unit
+
+    /**
+     * Called when NavigationView enters Free Drive state.
+     */
+    open fun onFreeDriveStarted() = Unit
+
+    /**
+     * Called when NavigationView enters Destination Preview state.
+     */
+    open fun onDestinationPreviewStared() = Unit
+
+    /**
+     * Called when NavigationView enters Route Preview state.
+     */
+    open fun onRoutePreviewStared() = Unit
+
+    /**
+     * Called when NavigationView enters Active Navigation state.
+     */
+    open fun onActiveNavigationStared() = Unit
+
+    /**
+     * Called when NavigationView enters Arrival state.
+     */
+    open fun onArrivalStared() = Unit
+
+    /**
+     * Called when Map [Style] has changed. Invoked once the new style has been fully loaded,
+     * including the style specified sprite and sources.
+     *
+     * @param style Fully loaded style.
+     */
+    open fun onMapStyleChanged(style: Style) = Unit
```

Comment:
I'm not sure about this one. We already expose the map instance, so we shouldn't wrap map APIs.

Reply:
Nope. We don't expose the map instance in NavigationView. We only allow the injection of a custom map instance via a binder.

Reply:
We should expose that instance. If we allow injecting a custom one, we should also give access to the existing one for developers to draw additional symbols, mark POIs etc. I remember us discussing maps usage in this context, instead of wrapping Maps SDK APIs that's been a pitfall in the past.

Reply:
I, Kyle and Tomasz had a brief discussion on this one. Based on our early discussion we had decided not to give access to `mapView`. However a few days back we revisited this discussion and we are on the same page to expose map view instance. I still have to cut a ticket for it. I don't take this as a blocker for this PR. This can be removed once we have exposed the map view instance.